apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/sampling/SamplingService.java

```diff
@@ -90,9 +88,9 @@ public void shutdown() {
      * @return true, if sampling mechanism is on, and getDefault the sampling factor successfully.
      */
     public boolean trySampling(String operationName) {
-        if (on) {
+        if (samplingRateWatcher.getSamplingRate() > 0) {
```

Comment:
Two `samplingRateWatcher.getSamplingRate() > 0`(s) in `#trySampling` and `#forceSampled` are changing the previous logic.
We don't need to `#getSamplingRate` every time, it costs resource. `#shouldSample` method needs to be added, and you should only change the boolean result when change happens.